packages/core/core/src/Environment.js

```diff
@@ -20,18 +20,20 @@ type EnvironmentOpts = {|
   loc?: ?InternalSourceLocation,
 |};
 
-export function createEnvironment({
-  context,
-  engines,
-  includeNodeModules,
-  outputFormat,
-  sourceType = 'module',
-  shouldOptimize = false,
-  isLibrary = false,
-  shouldScopeHoist = false,
-  sourceMap,
-  loc,
-}: EnvironmentOpts = {}): Environment {
+export function createEnvironment(
+  {
+    context,
+    engines,
+    includeNodeModules,
+    outputFormat,
+    sourceType = 'module',
+    shouldOptimize = false,
+    isLibrary = false,
+    shouldScopeHoist = false,
+    sourceMap,
+    loc,
+  }: EnvironmentOpts = {...null},
```

Comment:
This is weird. What's going on here? All the properties are optional right, so why null?

Reply:
This is not about the properties, it's literally only about exact/inexact/unsealed objects: https://github.com/facebook/flow/issues/7424

https://flow.org/try/#0DYUwLgBAHgXBDeAfRBfCBeBKDcAoUkAnnPGpqXgRAF4nJkIB0zAdgK7DA5A

Reply:
That's super weird when all the properties are optional. Therefore an empty object should be valid. I guess `{...null}` has some additional runtime overhead. _sigh_

Reply:
It gets weirder: https://flow.org/try/#0DYUwLgBAHgXBDeAfAhgfjgOwK4FsBGIATogL4QC8CJA3BAPR0RgAWIEAxloYSBpAA6EA9nlA4AUJNCQAnnHjJMuAoTKV4NeoxZscyAOYBLdhCEAzCFgwBnEMlAATU3gBWIdmGtA

Reply:
lol wow